### ISSUE TITLE
HM: drop-down list of units?

### ISSUE THREAD

**Issue by luizaogs** [arch-kiosk/arch-kiosk-office#2518]
Would it be worth adding a drop-down menu with the available units? Or is that not necessary since I guess we can assume that one knows what unit one is looking for? I think it might be nice, but if it's too much work probably not worth the trouble.

**Comment by urapadmin** [comment]
The lab project is simply as barren as it gets. Yes, in the final ui environment (Q&V) you will choose a unit (or a locus for that matter) from a drop down or jump to a HM graph from a unit or a locus.

**Comment by luizaogs** [comment]
Ah I thought the HM Lab was the final UI. That makes more sense :)

**Comment by urapadmin** [comment]
Ja, I should presumably honour you guys with a few more words of explanation when I roll test stuff out. It was just so very late 🌔

**Comment by urapadmin** [comment]
We might want to stick to a stand alone version of the Harris Matrix feature (as an alternative to the Q&V embedding) and so I should give it the option to select a unit / locus right here.

**Comment by urapadmin** [comment]
also: I keep forgetting tags as identifiers. In the end we might not only want to colour all the loci of a unit that are tagged with something, we might also want to show all the loci that are tagged with something together in its own HM. That can span units.

**Comment by urapadmin** [comment]
since we need to open identifiers letft and right in the future, I should make a general selection component first and then use it here:

**Comment by urapadmin** [comment]
### Kiosk 1.5.23.4
- little in-between release using #2543 in the lab project to test it

**Comment by lbestock** [comment]
This sometimes redraws depending on which locus I use to get to the FA matrix?? Is that expected?

**Comment by lbestock** [comment]
Sometimes it produces what looks like a bad matrix even. This is with FA-058.
![image](https://github.com/arch-kiosk/arch-kiosk-office/assets/49291909/cb347c74-2304-43cc-bacf-5786c47bf7a5)

**Comment by urapadmin** [comment]
choosing a locus doesn't really have a meaning, currently. You get all the relations Kiosk can find for such a locus. It is possible that this comes out differently than asking Kiosk for all relationsships of all the loci of a unit. It will make sense once we decide what exactly we want to do when choosing a locus.

**Comment by urapadmin** [comment]
but as for the "bad" matrix: see #2521 and #2545

**Comment by luizaogs** [comment]
Does it make sense for me to test this now or should I wait for the stuff above to be addressed?

**Comment by urapadmin** [comment]
the issues above are unrelated. This here is having the opportunity to choose an identifier rather than type one in. 

I see already the usual issue with this in PASU but it will be worse in USTP: Just the identifier isn't really enough information. I suppose the unit is always necessary, too. And perhaps the site if it is the unit. So actually a third column showing the "parent identifier".

**Comment by luizaogs** [comment]
Ok well, the UI for choosing and adding the unit seems to be working well. But you seem to have found issues already with PASU so I guess this stays open?

**Comment by luizaogs** [comment]
Actually, wait. I don't understand what this means? I'm supposed to be able to search for stuff to add too? 

<img width="475" alt="Screenshot 2024-02-13 at 8 23 50 PM" src="https://github.com/arch-kiosk/arch-kiosk-office/assets/67970402/9265b505-d694-4855-b42e-111c9f99ff60">

**Comment by luizaogs** [comment]
Oh, sorry. I'm stupid. Typing in gives me the loci.
